content/docs/getting-started/futures.md

```diff
@@ -74,282 +127,105 @@ They cannot because they don't have the actual HTTP response. Instead, they
-`NotReady` when at least one inner future returned `NotReady`. Thus, the outer
-future will transition to a ready state once the inner future transitions to a
-ready state. In this case, the `NotReady` contract is already satisfied as the
-inner future will notify the executor when it becomes ready.
-
-Innermost futures, sometimes called "resources", are the ones responsible for
-notifying the executor. This is done by calling [`notify`] on the task returned
-by [`task::current()`].
-
-We will be exploring implementing resources and the task system in more depth in
-a later section. The key take away here is **do not return `NotReady` unless you
-got `NotReady` from an inner future**.
-
-# A More Complicated Future
-
-Let's look at a slightly more complicated future implementation. In this case, we
-will implement a future that takes a host name, does DNS resolution, then
-establishes a connection to the remote host. We assume a `resolve` function
-exists that looks like this:
+For now it's just important to know that futures have two associated types: `Item`
+and `Error`. `Item` is the type of the value that the `Future` will yield when it
+completes. `Error` is the type of Error that the `Future` may yield if there's an
+error before that causes the `Future` from being able to complete.
+
+Finally, `Future`s have one method named `poll`. We won't go into too much detail
+about `poll` in this section since you don't need to know about `poll` to use futures
+with combinators. The only thing to be aware for now is that `poll` is
+what the runtime will call in order to see if the `Future` is complete yet or not. If
+you're curious: `Async` is an enum with values `Ready(Item)` or `NotReady` which
+informs the runtime of if the future is complete or not.
+
+In a future section, we'll be implementing a `Future` from scratch including writing
+a poll function that properly informs the runtime when the future is complete.
+
+## Streams
+
+Streams are the iterator equivalent of futures. Instead of yielding a value at some
+point in the future, streams yield a collection of values each at some point in the
+future. Just like futures, you can use streams to represent a wide range of things
+as long as those things produce discrete values at different points sometime in the
+future. For instance:
+* **UI Events** caused by the user interacting with a GUI in different ways. When an
+  event happens the stream yields a different message to your app over time.
+* **Push Notifications from a server**. Sometimes a request/response model is not
+  what you need. A client can establish a notification stream with a server to be
+  able to receive messages from the server without specifically being requested.
+* **Incoming socket connections**. As different clients connect to a server, the
+  connections stream will yield socket connections.
+
+Unlike futures, streams never yield values when they're complete, but rather
```

Comment:
This sentence is a bit confusing. Should it be removed? The section's opening paragraph covers what is needed I think?